crates/pathfinder/src/bin/pathfinder.rs

```diff
@@ -29,10 +31,16 @@ async fn main() -> anyhow::Result<()> {
 
     permission_check(&config.data_directory)?;
 
+    let prometheus_handle = PrometheusBuilder::new()
+        .install_recorder()
+        .context("Creating Prometheus recorder")?;
```

Comment:
Is there something to init outside of the `config.monitoring_addr`?

Reply:
If I understood the question correctly - you're right, no need to init the prometheus recorder in the outer scope, I'll move it, especially that by default `metrics` uses a global no-op recorder in case no "real" global recorder is set and we want this when monitoring is disabled.

BTW: `metrics::register_*!` is not really unnecessary unless you want a metric to be available before its value is changed for the first time. We want to see `0`s initially, hence the `register_counter!`.

Reply:
I think seeing that zero makes sense, at least in the ... reverse RAII sense :) "initialization is resource acquisition". Ideally metrics, like every other part of code, would be something carried as Arc throughout the app, not this singleton business.